crates/deltalake-core/src/operations/drop_constraints.rs

```diff
@@ -0,0 +1,247 @@
+
+            let found_constraint = metadata.configuration.contains_key(&configuration_key);
+            if this.raise_if_not_exists && !found_constraint {
+                return Err(DeltaTableError::Generic(format!(
+                    "Constraint with name: {} doesn't exists",
+                    name
+                )));
+            } else if !this.raise_if_not_exists && !found_constraint {
+                return Ok(DeltaTable::new_with_state(this.log_store, this.snapshot));
+            }
+
+            metadata
+                .configuration
+                .remove(format!("delta.constraints.{}", name).as_str());
+
+            let mut contains_constraints: bool = false;
+            for key in metadata.configuration.keys() {
+                if key.starts_with("delta.constraints") {
+                    contains_constraints = true;
+                    break;
+                } else {
+                    continue;
+                }
+            }
+
+            let operational_parameters = HashMap::from_iter([("name".to_string(), json!(&name))]);
+
+            let operations = DeltaOperation::DropConstraint { name: name.clone() };
+
+            let app_metadata = match this.app_metadata {
+                Some(metadata) => metadata,
+                None => HashMap::default(),
+            };
+
+            let commit_info = CommitInfo {
+                timestamp: Some(Utc::now().timestamp_millis()),
+                operation: Some(operations.name().to_string()),
+                operation_parameters: Some(operational_parameters),
+                read_version: Some(this.snapshot.version()),
+                isolation_level: Some(IsolationLevel::Serializable),
+                is_blind_append: Some(false),
+                info: app_metadata,
+                ..Default::default()
+            };
+
+            let protocol = if !contains_constraints {
+                // downgrade the protocol if possible since there are no more constraints
+                let old_protocol = this.snapshot.protocol();
+                Some(Protocol {
+                    min_reader_version: if old_protocol.min_reader_version > 1 {
+                        old_protocol.min_reader_version
+                    } else {
+                        1
+                    },
+                    min_writer_version: if old_protocol.min_writer_version > 3 {
+                        old_protocol.min_writer_version
+                    } else {
+                        2
+                    },
+                    reader_features: old_protocol.reader_features.clone(),
+                    writer_features: old_protocol.writer_features.clone(),
+                })
+            } else {
+                None
+            };
+
+            let mut actions = vec![Action::CommitInfo(commit_info), Action::Metadata(metadata)];
+
+            protocol.map(|protocol| actions.push(Action::Protocol(protocol)));
```

Comment:
while it's more code a `if let Some(protocol) = protocol {` makes it clearer on what you are intending here.